blueprint.js

```diff
@@ -198,12 +198,25 @@ $Element.prototype.copy = function() {
 			res.childNodes.push(child.clone());
 		}
 	}
+	res.data = Object.assign({}, this.data);
 	// Return the copy
 	return res;
 }
 
+$Element.prototype.prepare = function(callback) {
+	this.data.prepare = callback;
+	return this;
+}
+
 $Element.prototype.create =
-$Element.prototype.element = function() {
+$Element.prototype.element = function(props) {
+	if (this.data.prepare !== undefined) {
+		return this.data.prepare(this.copy(), props)._createElement();
```

Comment:
Ideally this would also change to not rely on the value returned by the preparation callback(s)

Reply:
That was mostly just me being lazy and thinking I could simplify things, but it is at the caller's expense.